packages/input_schema/src/types.ts

```diff
@@ -61,6 +61,19 @@ export type ArrayFieldDefinition = CommonFieldDefinition<Array<unknown>> & {
     items?: unknown;
 }
 
+export type CommonResourceFieldDefinition<T> = CommonFieldDefinition<T> & {
+    editor?: 'resourcePicker' | 'hidden';
+    resourceType: 'dataset' | 'keyValueStore' | 'requestQueue'
+}
+
+export type ResourceFieldDefinition = CommonResourceFieldDefinition<string> & {
+    type: 'string'
+}
+
+export type ResourceArrayFieldDefinition = CommonResourceFieldDefinition<Array<string>> & {
```

Comment:
i dont think we use `Array<T>` anywhere else, lets stay consistent

```suggestion
export type ResourceArrayFieldDefinition = CommonResourceFieldDefinition<string[]> & {
```